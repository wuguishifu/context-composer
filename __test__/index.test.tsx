import React from 'react';
import renderer from 'react-test-renderer';
import Composer from '../lib';
import { describe, it, expect } from '@jest/globals';

describe('Composer', () => {
    it('should render a div', () => {
        const component = renderer.create(
            <Composer components={[]}>
                <div id='child' />
            </Composer>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with providers', () => {
        const component = renderer.create(
            <Composer components={[TestProvider0, TestProvider1]}>
                <div id='child' />
            </Composer>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

function TestProvider0({ children }: { children: React.ReactNode }) {
    return <div id='p0'>{children}</div>;
}

function TestProvider1({ children }: { children: React.ReactNode }) {
    return <div id='p1'>{children}</div>;
}