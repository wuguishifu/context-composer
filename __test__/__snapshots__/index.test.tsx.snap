// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composer should render a div 1`] = `
<div
  id="child"
/>
`;

exports[`Composer should render with providers 1`] = `
<div
  id="p0"
>
  <div
    id="p1"
  >
    <div
      id="child"
    />
  </div>
</div>
`;
